### ISSUE TITLE
Feature Request support for RP2040-PiZero

### ISSUE THREAD

**Issue by actraiser30** [xrip/pico-megadrive#1]
Hello 

Will you please add support for the RP2040-PiZero?

https://www.waveshare.com/wiki/RP2040-PiZero

Thanks

**Comment by DnCraptor** [comment]
I guess, - not. It is for RP Pi **Pico** only.

**Comment by actraiser30** [comment]
That's a pico board also...

**Comment by xrip** [comment]
Seems it's possible by just change GPIO pins definitions.  I'll check this when have spare time

**Comment by DnCraptor** [comment]
> That's a rp pico board also...

no, it is "pi", but not "pico"

**Comment by xrip** [comment]
> > That's a rp pico board also...
> 
> no, it is "pi", but not "pico"

It's called RP2040-PiZero - RP2040 means something :)

**Comment by DnCraptor** [comment]
Sorry, mixed with other "Zero" https://www.cytron.io/p-raspberry-pi-zero-w

**Comment by actraiser30** [comment]
Yes thanks,it's all in one board, eliminating the need to buy extra things
to make it function able.

**Comment by DynaMight1124** [comment]
I've had a go at this as I have a RP2040 PiZero. Using SDK2.0 with VSCode and after some issues compiling the audio_i2s.pio file, I got it to compile without errors. Setup the pins for the SD, HDMI and NES port, after some messing with the HDMI pins I got a clear menu and it shows the roms, however its not working properly. It seemingly crashes quite often, sometimes boots to the menu, sometimes doesnt, if the menu boots it'll crash when trying to boot a game.

I'm not sure if this is an issue with the compiler, the PiZero or something else! Not been able to test anything like sound but when it does work, the NES controller works as does the SD card access so I'll upload the uf2 file and also the amended code for archiving sake.

Would be awesome if the MURMULATOR line also included support for devices like the RP2040 PiZero since its a very cost effective all in one device with plenty of GPIO.

[genesis-HDMI-RP2040-PiZero.zip](https://github.com/user-attachments/files/19814391/genesis-HDMI-RP2040-PiZero.zip)

**Comment by DnCraptor** [comment]
@DynaMight1124 , thnx for your initial investigation.
FYI: we have some movements to RP2040-PiZero side. Initial project, started such support - murmulator-tester:
https://github.com/DnCraptor/murmulator-tester
It uses slighly other pin-out, used by you.
Also, pls. do not miss, that original Raspberry Pi Pico uses GP25 as LED-out, and this GP in HDMI space for PiZero, so it may (should)) conflict.

**Comment by DnCraptor** [comment]
@DynaMight1124 , also, you may review
https://github.com/DnCraptor/pico-nes - this project that supports PiZero as well.
On my board it is not working with 378MHz, and I was forced to use 252 MHz instead

**Comment by DynaMight1124** [comment]
@DnCraptor thats awesome. Strangely wouldnt compile using the latest toolchain but 2.1.1 with 13_3_Rel1 worked fine (I got errors compiling xinput with latest toolchain) however for some reason I dont get any display after flashing. I assume its meant to have a frontend? or is it meant to be used with the Murmulator OS?

For Pico Megadrive, if the PiZero is struggling with higher pico clocks, could that had been a reason for the crashing? Not sure what the default clocks are? I understand that that means maybe it'll never work due to the high clocks needed for MD :)

**Comment by DnCraptor** [comment]
@DynaMight1124 , https://github.com/xrip/pico-megadrive/blob/main/src/main.cpp#L206 - a list of CPU freqs. You may try to inject 252 as first element.
Just FYI: compiled today with 2.1.1 and at least text mode passed for me on RP2040/RP2350 with Murmulator board. PiZeo case is not yet tested. My experiments are pushed to https://github.com/DnCraptor/pico-megadrive
@xrip, are you tested your latest master? It looks not fine after flashing, unable to find .md-file that works (and unable to switrch to start+select menu)
